src/components/input/input.js

```diff
@@ -650,7 +651,21 @@ function mdMaxlengthDirective($animate, $mdUtil) {
       // Using the $validators for triggering the update works very well.
       renderCharCount();
 
-      return ( modelValue || element.val() || viewValue || '' ).length <= maxlength;
+      // Force the value into a string since it may be a number,
+      // which does not have a length property.
+      return String(modelValue || element.val() || viewValue || '').length <= maxlength;
```

Comment:
1. This will not work correctly if `element.val()` is the number `0`.

2. What about an empty `modelValue`? If the intention is to validate on the `viewValue` (which would be is a reasonable approach; this is what AngularJS moves towards), then why take a look at the `modelValue` at all.

3. What about `ngTrim`? The char count would take that into account (and display the correct count, but the validator might report something that does not match the char count.